src/app/components/ContactToolbar.js

```diff
@@ -4,24 +4,34 @@ import { Icon, Checkbox } from 'react-components';
 import { c } from 'ttag';
 import ContactGroupDropdown from './ContactGroupDropdown';
 
-const ContactToolbar = ({ user, onCheck, onDelete, checked, checkedContacts, contactEmailsMap }) => {
+const ContactToolbar = ({ user, onCheck, onDelete, checked = false, contactID, checkedContacts, contactEmailsMap }) => {
+    // Include current contact as selected if none is checked
+    const activeContacts =
+        !!Object.values(checkedContacts).filter(Boolean).length || !contactID ? checkedContacts : { [contactID]: true };
+
     const handleCheck = ({ target }) => onCheck(target.checked);
 
     const contactEmailsSelected = useMemo(() => {
```

Comment:
Could we move `contactEmailsSelected` out to `ContactsContainer` and use it in `handleDelete` and pass the selected contacts here as well so that they are using the same logic?

Reply:
Indeed. The component was looking like robocop...

Reply:
Unfortunately I think you have to `useMemo` the activeIDs instead of calculating them on the fly due to performance in both places